src/cargo/ops/cargo_package.rs

```diff
@@ -406,25 +417,27 @@ fn check_for_file_and_add(
     Ok(())
 }
 
-fn warn_on_nonexistent_file(
+fn error_on_nonexistent_file(
     pkg: &Package,
     path: &Path,
     manifest_key_name: &'static str,
-    ws: &Workspace<'_>,
-) -> CargoResult<()> {
+    invalid: &mut Vec<String>,
+) {
     let rel_msg = if path.is_absolute() {
         "".to_string()
     } else {
         format!(" (relative to `{}`)", pkg.root().display())
     };
-    ws.gctx().shell().warn(&format!(
+
+    let msg = format!(
```

Comment:
Cargo checks only two fields, so it is fine just returning an `Err` here . That would make the code easier to reason.

Reply:
It would certainly be possible to return `Err` directly in the function `error_on_nonexistent_file`, but I think if neither `licence` nor `readme` exists, then they should all be prompted at the same time, not one by one.

And it's also easier to expand the other options later on as required fields as well. I hope my understanding is reasonable, thank you.